### PR TITLE
Add platfrom sequence telemetry param

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

See spec [here](https://microsoft-my.sharepoint-df.com/:w:/r/personal/oldalton_microsoft_com1/_layouts/15/doc2.aspx?sourcedoc=%7B6d81623f-6852-4b72-9a53-48062fc26eaa%7D&action=edit&wdPid=27527803&share=IQE_YoFtUmhyS5pTSAYvwm6qAY3fncu83FQi9DHz8uLLrIo).

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m [hunk 1]
```diff
@@ -73,4 +73,54 @@ - (NSDictionary *)msidParsedClientTelemetry
...
+        [sequenceItems addObject:@""]; //Browser-Ext-Ver
+        [sequenceItems addObject:@""]; //BrowserCore-Ver
+    }
+    else
+    {
+        sequenceItems = [[sequence componentsSeparatedByString:kDelimeter] mutableCopy];
+    }
+    
+    // Validate count.
+    if (sequenceItems.count != kSequenceItemsCount)
```
Comment (oldalton) [inline IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:98]: Should this be >= in case we increase count in the future?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m [hunk 2]
```diff
@@ -73,4 +73,54 @@ - (NSDictionary *)msidParsedClientTelemetry
...
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError,nil, @"Failed to add add platfrom sequence param: sequence count %d is invalid.", kSequenceItemsCount);
```
Comment (ameyapat) [inline IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:100]:
```suggestion
        MSID_LOG_WITH_CTX(MSIDLogLevelError,nil, @"Failed to add platfrom sequence param: sequence count %d is invalid.", kSequenceItemsCount);
```

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m [hunk 3]
```diff
@@ -73,4 +73,54 @@ - (NSDictionary *)msidParsedClientTelemetry
...
+        return nil;
+    }
+    
+    // Validate name.
+    if ([NSString msidIsStringNilOrBlank:name])
```
Comment (ameyapat) [inline IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:105]: Like 'Msal-Runtime-Ver' What if the sequenceItem is defined only by a version and has no name?

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m [hunk 4]
```diff
@@ -73,4 +73,54 @@ - (NSDictionary *)msidParsedClientTelemetry
...
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelWarning,nil, @"Failed to add add platfrom sequence param: name is empty/nil.");
+        return sequence;
+    }
+    
+    // Validate version.
+    if ([NSString msidIsStringNilOrBlank:version])
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelWarning,nil, @"Failed to add add platfrom sequence param: version is empty/nil.");
```
Comment (ameyapat) [inline IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:114]:
```suggestion
        MSID_LOG_WITH_CTX(MSIDLogLevelWarning,nil, @"Failed to add platfrom sequence param: version is empty/nil.");
```

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:98]: Good point, fixed.

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m [hunk 5]
```diff
@@ -73,4 +73,54 @@ - (NSDictionary *)msidParsedClientTelemetry
     return telemetryDict;
 }
 
++ (NSString *)msidAddPlatformSequenceParamWithName:(NSString *)name version:(NSString *)version toSequence:(NSString *)sequence
+{
+    NSMutableArray *sequenceItems;
+    static const int kSequenceItemsCount = 6;
+    static NSString *const kDelimeter = @",";
+
```
[the remark this answers is not in the document]
Comment (antrix1989) [reply IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:81]: Added enum

File: IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/src/util/NSString+MSIDTelemetryExtensions.m:105]: We only support first 2 params in this method. Msal runtime will need to created their own logic.

**Pull request merged**